pysatMissions/methods/spacecraft.py

```diff
@@ -173,18 +163,85 @@ def project_ecef_vector_onto_sc(inst, x_label, y_label, z_label,
 
     # TODO(#65): add checks for existence of ecef labels in inst
 
-    x, y, z = OMMBV.vector.project_onto_basis(
-        inst[x_label], inst[y_label], inst[z_label],
-        inst['sc_xhat_ecef_x'], inst['sc_xhat_ecef_y'], inst['sc_xhat_ecef_z'],
-        inst['sc_yhat_ecef_x'], inst['sc_yhat_ecef_y'], inst['sc_yhat_ecef_z'],
-        inst['sc_zhat_ecef_x'], inst['sc_zhat_ecef_y'], inst['sc_zhat_ecef_z'])
-    inst[new_x_label] = x
-    inst[new_y_label] = y
-    inst[new_z_label] = z
+    xx = inst['sc_xhat_ecef_x']
+    xy = inst['sc_xhat_ecef_y']
+    xz = inst['sc_xhat_ecef_z']
+
+    yx = inst['sc_yhat_ecef_x']
+    yy = inst['sc_yhat_ecef_y']
+    yz = inst['sc_yhat_ecef_z']
+
+    zx = inst['sc_zhat_ecef_x']
+    zy = inst['sc_zhat_ecef_y']
+    zz = inst['sc_zhat_ecef_z']
+
+    inst[new_x_label] = inst[x_label] * xx + inst[y_label] * xy + inst[z_label] * xz
+    inst[new_y_label] = inst[x_label] * yx + inst[y_label] * yy + inst[z_label] * yz
+    inst[new_z_label] = inst[x_label] * zx + inst[y_label] * zy + inst[z_label] * zz
 
     if meta is not None:
         inst.meta[new_x_label] = meta[0]
         inst.meta[new_y_label] = meta[1]
         inst.meta[new_z_label] = meta[2]
 
     return
+
+
+def normalize(x, y, z):
+    """Normalize a time-series of vectors.
+
+    Parameters
+    ----------
+    x : pds.Series
+        The x-component
+    y : pds.Series
+        The y-component
+    z : pds.Series
+        The z-component
+
+    Returns
+    -------
+    xhat : pds.Series
+        The normalized x-component
+    yhat : pds.Series
+        The normalized y-component
+    zhat : pds.Series
+        The normalized z-component
+
+    """
+
+    vector = np.vstack([x, y, z])
+    xhat, yhat, zhat = vector / np.linalg.norm(vector, axis=0)
+
+    return xhat, yhat, zhat
+
+
+def cross_product(x1, y1, z1, x2, y2, z2):
```

Comment:
How does this differ from `numpy.cross`?

Reply:
This works for a time series.  I could use numpy.cross, but would have to construct 2D variables each time (as in the normalize function).

Reply:
ie, I could change the code below to use `numpy.cross`, but the function would still be needed here.